SwitchSpeak-iOS/SwitchSpeak-iOS/ViewController.swift

```diff
@@ -5,198 +5,37 @@
 //  Created by Robert Gerdisch on 2/5/18.
 //  Copyright © 2018 SwitchSpeak. All rights reserved.
 //
-
-
 import UIKit
 
-class stash {
-	
-	var items: [item]
-	
-	init() {
-		self.items = []
-		
-	}
-	
-	
-	func add( new_item:item ){
-		items.append(new_item)
-		new_item.Set_Loc_Col(index:items.count-1 )
-	}
-	
-	
-	func delete(){
-		items.removeLast()
-	}
-	
-	func add(new_item : String) {
-		let newitem = item(content: new_item)
-		add(new_item: newitem)
-	}
-	
-	
-	
-}
-
-class item {
-	var button : UIButton
-	
-	init(content : String) {
-		button = UIButton(type: UIButtonType.system)
-		button.setTitle(content, for: UIControlState.normal)
-		//  button.backgroundColor = UIColor.darkGray
-		
-	}
-	
-	func Set_Loc_Col(index : Int ){
-		button.frame = CGRect(x:10+(index % 5)*80, y:40*(index/5+1), width:80, height:40)
-		if index % 2 == 0{
-			button.backgroundColor = UIColor.darkGray
-			
-		}
-		else{
-			button.backgroundColor = UIColor.lightGray
-		}
-		
-	}
-	
-}
-
 class ViewController: UIViewController {
-	
-	
 	@IBOutlet weak var TapButton: UIButton!
-	
-	var curNode = Node()	//	currently the node in the tree we are at while scanning
-	var childNumber: Int = 0 //	this will represent the index of highlighted child of the curNode
-	//	the result of the scanning procedure will be stored
-	//	in curNode and childNumber variables
-	var flag: Bool = false	//	this is set to true whenever curNode is updated
-	var breadcrum = stash()
-	var factory = TreeFactory()
-	var T = Tree()
-	
-	//	the following attributes can be input from user settings
-	var timeDelay: Double = 0.8	//	time delay during scanning
-	var rows = 3, cols = 4	//	dimension of the 2D grid
-	
-	
+	var breadcrumbs = CrumbStack()
+    var touchGrid:TouchGrid?
 	
 	override func viewDidLoad() {
 		super.viewDidLoad()
-		self.T = factory.treeForRowColumnScanning(rows: rows, cols: cols)
-		
-		//	there are three magic numbers in the below function call
-		//	need to modify these magic numbers to appropriate variables
-		self.T.setFrameSizeForTree (row: rows, col: cols, screenWidth: Int(UIScreen.main.bounds.width), screenHeight: Int(UIScreen.main.bounds.height), topBarHeight: 140, maxButtonHeight: 200, maxButtonWidth: 200)
-		
-		addTreeToView(T: T)
-		view.bringSubview(toFront: TapButton)
-		curNode = T.rootNode!
-		selectSubTree()
-		
+        
+        // Eventually '0' here would be replaced by the id of the 'signed-in' user
+        self.touchGrid = TouchGrid(userId: 0, viewContainer: view)
```

Comment:
we may add one more parameter for the TouchGrid function, which will represent an array of buttons (containing phrases and images); and this array can be used to initialize the grid.

Reply:
Good idea. I'm going to save this for a separate PR when we integrate the database.